config/admin-functions/create-machine-cert.sh

```diff
@@ -79,38 +79,20 @@ function wait_for_usb_and_mount_once_present() {
 
 # OpenSSL helpers
 
-function generate_password_with_256_bits_of_entropy() {
-    # 32 bytes * 8 bits per byte = 256 bits of entropy
-    openssl rand -base64 32
-}
-
-function generate_private_key() {
-    local private_key_path="${1}"
-    local private_key_password="${2}"
-    openssl ecparam -genkey -name prime256v1 -noout | \
-        openssl pkcs8 -topk8 -passout "pass:${private_key_password}" -out "${private_key_path}"
-}
-
-function construct_machine_cert_subject() {
-  local machine_jurisdiction="${1:-}"
-  if [[ -n "${machine_jurisdiction}" ]]; then
-    echo "/C=US/ST=CA/O=VotingWorks/${VX_IANA_ENTERPRISE_OID}.1=${VX_MACHINE_TYPE}/${VX_IANA_ENTERPRISE_OID}.2=${machine_jurisdiction}/"
-  else
-    echo "/C=US/ST=CA/O=VotingWorks/${VX_IANA_ENTERPRISE_OID}.1=${VX_MACHINE_TYPE}/"
-  fi
-}
-
 function create_machine_cert_signing_request() {
-    local machine_private_key_path="${1}"
-    local machine_private_key_password="${2}"
-    local machine_cert_subject="${3}"
-    local cert_signing_request_path="${4}"
-    openssl req -new \
-        -config "${VX_METADATA_ROOT}/vxsuite/libs/auth/certs/openssl.cnf" \
-        -key "${MACHINE_PRIVATE_KEY_PATH}" \
-        -passin "pass:${machine_private_key_password}" \
-        -subj "${machine_cert_subject}" \
-        -out "${cert_signing_request_path}"
+    local machine_jurisdiction="${1:-}"
+    if [[ -n "${machine_jurisdiction}" ]]; then
+        # Use a subshell to ensure the cd doesn't have an effect beyond this command
+        (cd "${VX_METADATA_ROOT}/vxsuite/libs/auth" &&
```

Comment:
you could also use `pushd` and `popd` if you wanted, but I think a subshell is fine?

Reply:
Nice! I like `pushd` and `popd` for clarity without an explanatory comment: https://github.com/votingworks/vxsuite-complete-system/pull/302/commits/43cff6f94ba3b2ee102c04869c5c8bc0b1de1f0c